include/zephyr/sys/fdtable.h

```diff
@@ -7,13 +7,33 @@
 #define ZEPHYR_INCLUDE_SYS_FDTABLE_H_
 
 #include <stdarg.h>
-#include <sys/types.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
 
-/* FIXME: For native_posix ssize_t, off_t. */
-#include <zephyr/fs/fs.h>
 #include <zephyr/kernel.h>
 #include <zephyr/sys/util.h>
 
+/*
+ * Note: avoid using POSIX headers and types in layers below POSIX.
+ * e.g. <sys/types.h>, ssize_t, off_t.
+ */
+
+#ifndef _SSIZE_T_DECLARED
```

Comment:
This guard is what pico (and newlib) use, but other libcs may have other guards (already minimal in tree)

Reply:
Can you point out a specific build that would be broken?